sstables/sstables.cc

```diff
@@ -2564,10 +2565,16 @@ future<uint32_t> sstable::read_digest() {
     co_return boost::lexical_cast<uint32_t>(digest_str);
 }
 
-future<checksum> sstable::read_checksum() {
+future<> sstable::read_checksum() {
+    if (_components->checksum) {
+        co_return;
+    }
+    co_await read_toc();
```

Comment:
Why do we have to re-read the TOC?

Reply:
It's a no-op if called more than once and we also do the same thing for some other components (`read_summary()`, `read_scylla_metadata()`).

For the current use cases of `read_checksum()` I'm pretty sure it's redundant so I'll remove it.